access/src/main/java/edu/unc/lib/dl/ui/controller/FedoraContentController.java

```diff
@@ -87,6 +92,11 @@ private void streamData(String pidString, String datastream, boolean asAttachmen
         PID pid = PIDs.get(pidString);
         AccessGroupSet principals = getAgentPrincipals().getPrincipals();
 
+        RepositoryObject updatedPid = repoObjLoader.getRepositoryObject(pid);
+        if (updatedPid.getClass() == WorkObject.class) {
+            pid = ((WorkObject) updatedPid).getPrimaryObject().getPid();
```

Comment:
not all works are guaranteed to have a primary object, so you should check that its null before getting the pid